multicluster/controllers/multicluster/internal/remote_cluster.go

```diff
@@ -0,0 +1,300 @@
+package internal
+
+import (
+	multiclusterv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"strings"
+	"sync"
+	"time"
+)
+
+var (
+	// Use a raw client to get member cluster secrets from antrea-plus-system only.
+	secretCient client.Client
+	// multiple remote clusters can be created concurrently, need a lock to synchronize creation of secretClient
+	mutex sync.Mutex
+)
+
+type RemoteCluster interface {
+	common.CommonArea
+
+	Start() (context.CancelFunc, error)
```

Comment:
Probably add some comments to describe what RemoteCluster will do in the started routine.